src/frontend/hooks/useMapStyle.ts

```diff
@@ -0,0 +1,151 @@
+import * as React from "react";
+import MapboxGL from "@react-native-mapbox-gl/maps";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import ky from "ky";
+import config from "react-native-build-config";
+
+import api from "../api";
+import useWifiStatus from "./useWifiStatus";
+import { normalizeStyleURL } from "../lib/mapbox";
+import { useExperiments } from "./useExperiments";
+
+/** Key used to store most recent map id in Async Storage */
+const MAP_STYLE_KEY = "@MAPSTYLE";
+
+/** URL used for map style when no custom map and user is online */
+const onlineStyleURL = MapboxGL.StyleURL.Outdoors + "?" + Date.now();
+
+/** URL used for map style when user is not online
+ * generated by [mapeo-offline-map](https://github.com/digidem/mapeo-offline-map) */
+const fallbackStyleURL = "asset://offline-style.json";
+
+type MapActions =
+  | { type: "setCustomLegacy" | "setMapServer"; style: string }
+  | { type: "setLoading" | "setOnline" | "setFallback" };
+
+export type MapStyleType =
+  | "customLegacy"
+  | "online"
+  | "fallback"
+  | "mapServer"
+  | "loading";
+
+type MapStyleState = {
+  mapStyle: MapStyleType;
+  mapStyleURL: string | null;
+};
+
+function mapStyleReducer(
+  state: MapStyleState,
+  action: MapActions
+): MapStyleState {
+  switch (action.type) {
+    case "setCustomLegacy":
+      return {
+        mapStyle: "customLegacy",
+        mapStyleURL: action.style,
+      };
+    case "setOnline":
+      return {
+        mapStyle: "online",
+        mapStyleURL: onlineStyleURL,
+      };
+    case "setFallback":
+      return {
+        mapStyle: "fallback",
+        mapStyleURL: fallbackStyleURL,
+      };
+    case "setMapServer":
+      return {
+        mapStyle: "mapServer",
+        mapStyleURL: action.style,
+      };
+    default:
+      return {
+        mapStyle: "loading",
+        mapStyleURL: null,
+      };
+  }
+}
+
+/**
+ * @param {string} styleId
+ * @returns `[mapStyle, setMapStyle(styleId:string)]` setMapStyle is a function that can be used to set the map style when map server is on
+ */
```

Comment:
nit: probably not necessary to do a full jsdoc annotation for ts files. if you just want a description, i think a multiline comment with the text will work i.e.

```
/**
 * Description of function
 */
```